### PR TITLE
Add MFEM_MPIEXEC config var to set alternate MPI launcher [alternate_mpi_launcher]

### PR DESCRIPTION
**Pull request by aaroncblack**
Allows setting an alternate command for launching MPI jobs, for systems that use SLURM or other job schedulers instead of 'mpirun'.

### CODE REVIEW

**Comment by aaroncblack** [comment]
@trb413 - I need to add CMAKE logic for this too...

**Comment by v-dobrev** [comment]
I think `MFEM_MPIRUN` is a more consistent name for this variable, cf. `MFEM_CXX`.

File: config/cmake/MFEMConfig.cmake.in [hunk 1]
```diff
@@ -11,7 +11,10 @@
 
 include(${CMAKE_CURRENT_LIST_DIR}/MFEMConfigVersion.cmake)
 
+set(MFEM_MPIRUN_CMD @MFEM_MPIRUN_CMD@)
```
Comment (v-dobrev) [inline config/cmake/MFEMConfig.cmake.in]: I don't think we need to have this variable here. CMake has its own variables for this, `MPIEXEC`, `MPIEXEC_NUMPROC_FLAG`, etc; see e.g. `examples/CMakeLists.txt`.

Comment (aaroncblack) [inline config/cmake/MFEMConfig.cmake.in]: So:
set(MFEM_MPIRUN $(MPIEXEC) $(MPIEXEC_NUMPROC_FLAG)) ?

Comment (aaroncblack) [inline config/cmake/MFEMConfig.cmake.in]: Perhaps splitting them ( like cmake ) will be the most flexible solution:
MFEM_MPIEXEC
MFEM_MPIEXEC_NUMPROC_FLAG

@v-dobrev - have a preference?

Comment (v-dobrev) [inline config/cmake/MFEMConfig.cmake.in]: Maybe I'm missing something but I don't see a reason to define any MPI related variables in `MFEMConfig.cmake`.

For the purposes of testing mfem with cmake/ctest, we already use the "standard" cmake variables `MPIEXEC`, `MPIEXEC_NUMPROC_FLAG`, etc. That way, a user can control the behavior using the "standard" MPI cmake variables.

File: examples/makefile [hunk 2]
```diff
@@ -51,23 +51,23 @@ include $(TEST_MK)
 
 # Testing: Parallel vs. serial runs
 %-test-par: %
-	@$(call mfem-test,$<, mpirun -np 4, Parallel example)
+	@$(call mfem-test,$<, $(MFEM_MPIEXEC) $(MFEM_MPIEXEC_NUMPROC_FLAG) $(MFEM_MPIEXEC) $(MFEM_MPIEXEC_NUMPROC_FLAG) 4, Parallel example)
```
Comment (aaroncblack) [inline examples/makefile]: will fix this line

**Comment by aaroncblack** [comment]
Renamed this branch.  The '/' screws up some of the test scripts.

**Pull request closed without merging**